### PR TITLE
Adds `tarmak cluster kubeconfig` subcommand

### LINKED ISSUE #279: Provide a kubeconfig subcommand for clusters
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

People/CI jobs tend to use other CLI tools that read the kubernetes config themselves and a kubeconfig could be a good way to make that more approachable.

**What you expected to happen**:

A kubeconfig subcommand should verify that the cluster is reachable (similar like kubectl subcommand is doing) and output the path to the kubeconfig like that:

```
KUBECONFIG=/full/path/to/kubeconfig
```

It then can be used in a shell session like that:

```
eval "$(tarmak cluster kubeconfig)"
kubectl get nodes
helm list
```

We also should support a public flag that will behave slightly different:

* Error if .kubernetes.apiserver.public is false
* Write config to `kubeconfig-public` and use the ELB public URL instead

### PR DESCRIPTION
**Pull request by JoshVanL**
**What this PR does / why we need it**:
Adds `tarmak cluster kubeconfig` sub command to verify and create the kubeconfig file.
Supports use of eval "$(tarmak cluster kubeconfig)" to set the KUBECONFIG environment variable.
Can set output path with --path flag. Default is default kubeconfig path.
Can set kubeconfig to point to the public API endpoint.

fixes #279 

```release-note
Adds `tarmak cluster kubeconfig`
```

/assign @simonswine

### CODE REVIEW

File: cmd/tarmak/cmd/cluster.go [hunk 1]
```diff
@@ -95,6 +95,25 @@ func clusterFlagDryRun(fs *flag.FlagSet, store *bool) {
 	)
 }
 
+func clusterKubeconfigFlags(fs *flag.FlagSet) {
+	store := &globalFlags.Cluster.Kubeconfig
+
+	fs.BoolVar(
+		&store.PublicAPIEndpoint,
+		"public-api-endpoint",
+		false,
```
Comment (simonswine) [inline cmd/tarmak/cmd/cluster.go]: This should be true for clusters where there is a public API endpoint available, false if not

Both values should be able to be overriden

**Comment by JoshVanL** [comment]
/unassign
@simonswine behavior should be better now

**Comment by simonswine** [comment]
Thank you a few more thoughts:

- Should we have the same public-api-server flag behaviour on kubectl (would probably mean we use it as a global flag instead)?
- I would like to have that example usage in the docs:

```
eval "$(tarmak cluster kubeconfig)"
kubectl get nodes
helm install
```

/assign @JoshVanL 
/unassign

**Comment by JoshVanL** [comment]
@simonswine Yep, guess that will have to be global :+1:

**Comment by simonswine** [comment]
I think this is almost ready:

```
tarmak kubeconfig --public-api-endpoint=true
DEBU[0000] check SSH connection to bastion cmd=[ssh -F /home/christian/.tarmak-dev/csire-cluster/ssh_config bastion /bin/true]  app=tarmak destination=api.csire-cluster.tarmak.local
DEBU[0000] start tunnel cmd=[ssh -F /home/christian/.tarmak-dev/csire-cluster/ssh_config -O forward -L127.0.0.1:41351:api.csire-cluster.tarmak.local:6443 bastion]  app=tarmak destination=api.csire-cluster.tarmak.local
DEBU[0000] trying to connect to https://api.csire-cluster.develop.tarmak.org  app=tarmak
DEBU[0000] connected to Kubernetes API v1.11.3           app=tarmak
KUBECONFIG=/home/christian/.tarmak-dev/csire-cluster/kubeconfig
```

I am not too sure why there is this tunnel start line. Ideally we would not need any tunnel for public APIs (apart from getting the vault cert). (It is the same for kubectl)

Are we starting a tunnel or not?

/assign @JoshVanL 
/unassign

**Comment by simonswine** [comment]
That is it thank you!

**Pull request merged**